File: HW1_Task3/src/com/company/Rank.java
package com.company;

public enum Rank {
    two(2), three(3), four(4), five(5), six(6), seven(7), eight(8), nine(9), ten(10), jack(11),
    quin(12), king(13), ace(14), joker(15);
    private final int rankValue;

    Rank(int rank) {
        rankValue = rank;
    }

    public int getRankValue() {
        return rankValue;
    }
}

File: HW3_Task1/HW3_Task1/src/Rank.kt
enum class Rank(rank: Int, name: String) {
    TWO(2, "two"), THREE(3, "three"), FOUR(4, "four"), FIVE(5, "five"), SIX(6, "six"), SEVEN(7, "seven"), EIGHT(8, "eight"),
    NINE(9, "nine"), TEN(10,"ten"), JACK(11, "jack"), QUIN(12,"quin"), KING(13,"king"), ACE(14, "ace"), JOKER(15, "joker");

    private val rankValue: Int = rank;
    private val rankName: String = name
    fun getRank(): Int {
        return rankValue
    }
    fun getRankName(): String {
        return rankName
    }
}